src/terra-dev-site/doc/application/how-to.2/theme-strategy.doc.mdx

```diff
@@ -0,0 +1,166 @@
+
+## Getting Started
+
+To get started, check out the [terra-application](http://engineering.cerner.com/terra-application/components/terra-application/application/reference/components/application-base) documentation. For this guide, it is expected that teams are utilizing an application framework that defines a theme provider. This guide will walk through theming an example component.
+
+## Theme Context
+
+An application's theme is provided through [React context](https://reactjs.org/docs/context.html) by a theme provider. [Application Base](http://engineering.cerner.com/terra-application/components/terra-application/application/reference/components/application-base) initializes a theme provider automatically. The theme value is accessed using the [theme context](http://engineering.cerner.com/terra-application/components/terra-application/application/reference/contexts/theme-context) from terra-application.
+
+```jsx
+import { ThemeContext } from 'terra-application/lib/theme';
+```
+
+Use the theme context to read the active theme value.
+
+```jsx
+import { ThemeContext } from 'terra-application/lib/theme';
+
+const ExampleComponent = () => {
+  // Access the active theme context value.
+  const theme = useContext(ThemeContext);
+};
+```
+
+The active theme name is stored in a `className` key in the context value. To apply the theme, bind the theme className to the component using [classnames](https://github.com/JedWatson/classnames). Some additional documentation for CSS Modules and classNames can be found on the [conventions page](https://engineering.cerner.com/terra-ui/contributing/terra-ui/conventions). Please note that an empty or undefined className indicates the default theme is active or a theme provider is missing. The `className` key maps directly to the anticipated theme name. Using the correct theme className is required.
+
+```jsx
+import React from 'react';
+import classNames from 'classnames/bind';
+import { ThemeContext } from 'terra-application/lib/theme';
+import styles from './ThemedComponent.module.scss';
+
+const cx = classNames.bind(styles);
+
+const ExampleComponent = () => {
+  const theme = React.useContext(ThemeContext);
+
+  return (
+    <div className={cx('example-component', theme.className)}>
+      This is an example component.
+    </div>
+  );
+};
+```
+
+## SCSS
+
+Once the theme className has been bound, the component can be appropriately themed. Start by defining the variables for the CSS properties that are expected to change between themes. SCSS variables are defined using [custom properties](https://developer.mozilla.org/en-US/docs/Web/CSS/--*). These values should utilize the SCSS [var()](https://developer.mozilla.org/en-US/docs/Web/CSS/var) functionality. The var() function works by applying a variable if the value has been set. If no value has been set the default value is used.
+
+`example-component/ExampleComponent.module.scss`
```

Comment:
```suggestion
### example-component/ExampleComponent.module.scss
```

Reply:
I'm still on the lookout for a good way to label code blocks. 

I don't think steps would be appropriate here for labeling. Headers are more ideal but throw a lint warning if more than two headers have the same text. I think because a unique anchor cannot be made for two identical headers. It's doesn't seem to be a concern in this markdown document but I run into the warning frequently when using headers as labels

Reply:
I'd use headers, will you have duplicate named headers on this page?

Reply:
I can tweak a few things so it doesn't happen on this document. Referencing the same file twice creates a duplicate header. Headers look nice visually, but I think fall short as labels for code blocks